Nodejs/Product/Nodejs/Project/NodejsGeneralPropertyPage.cs

```diff
@@ -95,32 +87,40 @@ protected override void Apply()
                 this.Project.SetUserProjectProperty(NodeProjectProperty.DebuggerPort, this.control.DebuggerPort);
             }
 
-            this.control.IsDirty = false;
+            this.IsDirty = false;
         }
 
         public override void LoadSettings()
         {
-            this.control.NodeExeArguments = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodeExeArguments);
-            this.control.NodeExePath = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodeExePath);
-            this.control.ScriptFile = this.Project.GetUnevaluatedProperty(CommonConstants.StartupFile);
-            this.control.ScriptArguments = this.Project.GetUnevaluatedProperty(NodeProjectProperty.ScriptArguments);
-            this.control.WorkingDirectory = this.Project.GetUnevaluatedProperty(CommonConstants.WorkingDirectory);
-            this.control.LaunchUrl = this.Project.GetUnevaluatedProperty(NodeProjectProperty.LaunchUrl);
-            this.control.NodejsPort = this.Project.GetUnevaluatedProperty(NodeProjectProperty.NodejsPort);
-            this.control.DebuggerPort = this.Project.GetUnevaluatedProperty(NodeProjectProperty.DebuggerPort);
-            this.control.Environment = this.Project.GetUnevaluatedProperty(NodeProjectProperty.Environment);
-            this.control.StartWebBrowser = GetBoolProperty(this.Project, NodeProjectProperty.StartWebBrowser);
-            this.control.SaveNodeSettingsInProject = GetBoolProperty(this.Project, NodeProjectProperty.SaveNodeJsSettingsInProjectFile);
-            this.control.TestRoot = this.Project.GetProjectProperty(NodeProjectProperty.TestRoot);
-            this.control.TestFramework = this.Project.GetProjectProperty(NodeProjectProperty.TestFramework);
+            this.Loading = true;
```

Comment:
>this.Loading = true; [](start = 12, length = 20)

Where is this used?

Reply:
There seems to be one reference to it in the base class.

https://github.com/microsoft/nodejstools/blob/0e0caef24ae5dcd9b76fbe57c617ec0ca9741ed4/Nodejs/Product/Nodejs/SharedProject/CommonPropertyPage.cs#L250

Reply:
In that case, is it thread-safe?

Reply:
Yes. VS UI controls all have UI thread affinity as far as I know. Confirmed via debugger.

Reply:
I feel like I'm missing something.  Does this method yield the UI thread so that another caller can check `IsDirty` on the UI thread?  If it just holds the UI thread until loading's done, the flag seems redundant?

Reply:
Yes, there's one thing that's not obvious from reading: setting the properties on the `control` is what causes the `IsDirty` setter to get called. This flag just seems to be for preventing excessive `OnStatusChanged` notifications from being sent as you initialize the page.